server/data/data-library/src/main/java/org/apache/james/dlp/eventsourcing/EventSourcingDLPConfigurationStore.java

```diff
@@ -60,29 +63,29 @@ public EventSourcingDLPConfigurationStore(EventStore eventStore) {
     }
 
     @Override
-    public DLPRules list(Domain domain) {
+    public Publisher<DLPRules> list(Domain domain) {
 
         DLPAggregateId aggregateId = new DLPAggregateId(domain);
 
-        return DLPDomainConfiguration.load(
-                aggregateId,
-                eventStore.getEventsOfAggregate(aggregateId))
-            .retrieveRules();
+        return Mono.from(eventStore.getEventsOfAggregate(aggregateId))
+            .map(history -> DLPDomainConfiguration.load(aggregateId, history).retrieveRules());
     }
 
     @Override
     public void store(Domain domain, DLPRules rules) {
-        eventSourcingSystem.dispatch(new StoreCommand(domain, rules));
+        Mono.from(eventSourcingSystem.dispatch(new StoreCommand(domain, rules))).block();
```

Comment:
idem for this method, and the `clear` and `fetch` one

Reply:
In another PR